lib/oli/delivery/gating/gating_condition.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Oli.Delivery.Gating.GatingCondition do
+  @moduledoc """
+  GatingCondition represents a condition which will be evaluated at delivery
+  to determine a user's ability to access a resource
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "gating_conditions" do
+    field :type, Ecto.Enum,
+      values: [
+        :start_datetime,
+        :end_datetime
+      ]
+
+    # data used by the condition evaluator, e.g. start or end datetime, a list of
+    # resource_ids, etc.
+    field :data, :map
```

Comment:
We probably want to model this more explicitly in the database, and model this as a "subtype" in a single table approach (similiar to how Revision has different types).  So instead of an opaque "data" blob, create specific fields for each type of GatingCondition.  For now, with only supporting scheduling, we would simply add a "date_time" field.  As we add in additional GatingCondition types, such as "required completed attempt of resouruce", we will add any additional fields.  The :type attribute informs the code which "subtype" this condition is, and thus which subset of fields to use.